### ISSUE TITLE
Xpetra:  No return from non-void

### ISSUE THREAD

**Issue by jmgate** [trilinos/Trilinos#1229]
I'm seeing the following when building with Xpetra_ENABLE_Experimental ON:
```
In file included from /build/Packages/Trilinos/packages/xpetra/src/CrsMatrix/Xpetra_CrsMatrixFactory.hpp:54:0,
                 from /build/Packages/Trilinos/packages/xpetra/sup/Matrix/Xpetra_Matrix.hpp:60,
                 from /build/Packages/Trilinos/packages/xpetra/sup/Utils/Xpetra_IteratorOps.hpp:52,
                 from /build/Packages/Trilinos/packages/xpetra/sup/Utils/Xpetra_IteratorOps.cpp:47:
/build/Packages/Trilinos/packages/xpetra/src/CrsMatrix/Xpetra_TpetraCrsMatrix.hpp: In member function ‘bool Xpetra::TpetraCrsMatrix<Scalar, int, long long int, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Serial> >::haveGlobalConstants() const’:
/build/Packages/Trilinos/packages/xpetra/src/CrsMatrix/Xpetra_TpetraCrsMatrix.hpp:1121:41: warning: no return statement in function returning non-void [-Wreturn-type]
     bool haveGlobalConstants() const  { }
                                         ^
```
I would think the behavior is undefined when there's no return from a function that should be returning a `bool`.  Is that not the case and it defaults to returning `false`?
@trilinos/xpetra

**Comment by tawiesn** [comment]
@csiefer2 I think we should return false.

I think there is no guarantee, that the compiler returns false by default...

**Comment by ibaned** [comment]
There is no default guarantee, the answer you get is whatever value was in the CPU register that was supposed to receive the return value but did not, so it will be random based on what code was executed prior to that.

**Comment by csiefer2** [comment]
Checkin test chugging.

**Comment by tawiesn** [comment]
fix is in develop branch (see commit 085bacda97fd)
